### PR TITLE
install: only support default prefix on Linux.

### PR DESCRIPTION
**Pull request by MikeMcQuaid**
The experience in a non-default experience is very poor. Let's not encourage it.

### CODE REVIEW

File: install.sh [hunk 1]
```diff
@@ -507,41 +509,15 @@ fi
...
-    fi
-    trap - SIGINT
-  fi
-  HOMEBREW_REPOSITORY="${HOMEBREW_PREFIX}/Homebrew"
+elif ! [[ -w "${HOMEBREW_PREFIX}" ]] &&
+     ! [[ -w "/home/linuxbrew" ]] &&
+     ! [[ -w "/home" ]] &&
+     ! have_sudo_access
+then
+  abort "Insufficient permissions to install Homebrew to \"${HOMEBREW_PREFIX}\"."
```
Comment (XuehaiPan) [inline install.sh:520]: > Let's not encourage it.

Should we add a link <https://docs.brew.sh/Installation#alternative-installs> here to tell people how to install Homebrew without `sudo` privilege? One of the reasons the user using Homebrew is they don't have `sudo` privilege.

**Review by carlocab — DISMISSED** [review]
I'd avoid pointing users to the alternative installs page for now, but consider adding it if we get multiple issues that could have been avoided if we point them there.

Alternatively, we should just link to our documentation on installation on any type of failure so help users work out what to do for themselves.

File: install.sh (hunk 1 above)
Comment (XuehaiPan) [reply install.sh]: > It'll be clearer that this is unsupported after Homebrew/brew#13971

I see, we only support default prefixes both on macOS and Linux now. If non-default prefixes are not officially supported, does that means we will not ship relocatable bottles in the future or the priority is lowered?

Comment (MikeMcQuaid) [reply install.sh]: @XuehaiPan We've only ever supported default prefix/bottles on Linux, we've just been bad at wording it.

If we can get to the point where we have roughly >=90% bottles relocatable outside the default prefix: we'll reconsider what we support.

Our support is pretty much based on "on what configurations can we continue to provide bottles".

**Pull request merged**